src/ensembl/src/shared/components/badged-button/BadgedButton.test.tsx

```diff
@@ -26,66 +26,42 @@ const defaultProps = {
   badgeContent: faker.lorem.words()
 };
 
-const renderButton = (
-  ButtonComponent: React.FunctionComponent<any>,
-  props: any = defaultProps
-) => mount(<ButtonComponent {...props} />);
+const renderButton = (props: any = {}) =>
+  render(
+    <BadgedButton {...defaultProps} {...props}>
+      <Button onClick={onClick}>{faker.lorem.words()}</Button>
+    </BadgedButton>
+  );
 
 describe('BadgedButton', () => {
-  let wrapper: any;
-
+  let container: any;
   beforeEach(() => {
-    wrapper = renderButton(BadgedButton, {
-      ...defaultProps,
-      children: <Button onClick={onClick}>{faker.lorem.words()}</Button>
-    });
-  });
-
-  afterEach(() => {
     jest.resetAllMocks();
+    container = renderButton().container;
```

Comment:
I would suggest running this line inside each `it`-block rather than in the `beforeEach`, for better flexibility, in the following form:

`const { container } = renderButton()`;

The reason for this is that the `render` function used in the `renderButton` returns an object with a number of useful fields besides `container`. For example, this object has a `rerender` method for re-rendering a component, and a `debug` method for pretty-printing the html representation of the component. You are losing access to those by plucking just the `container` from `renderButton`'s return value.

Reply:
if thats the case (we need the `rerender`  or `debug`) then we can just call the renderButton() like i did here https://github.com/Ensembl/ensembl-client/blob/762dbeaf9e218b30a6a23f21320f0d0cc01439d4/src/ensembl/src/shared/components/badged-button/BadgedButton.test.tsx#L52 . does that not work?  I quite like the fact i didnt had to rewrite the renderButton() everytime and was finding it confusing that it was the same code being repeated and if there was a way to have a const that will make the code clearer. And the part where you have rerender it is clear you are passing some props to rerender.
I can see the benefit of `debug` but i have used container.innerHTML a lot for debugging. 
Let me know what you think? I just thought the code looks more understandable :)

Reply:
> I quite like the fact i didnt had to rewrite the renderButton() everytime and was finding it confusing that it was the same code being repeated

Kent C. Dodds, the author of react-testing-library and a man of many opinions 😄, has written an [article](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing) advising against putting the variables referencing the rendered components globally. I think I agree with him, at least in the part about the actual rendering of a component. In our tests, it's just a single line anyway. Easy to repeat, easy to modify, easy to read within the context of a single test.

Reply:
I am not going to upset Kent C. Dodds :)